Model/Validate.php

```diff
@@ -103,6 +121,11 @@ public function beforeSave()
                 $this->activateSrrCustomerSegment($shopId, $shopPassword, $customerSegment["id"]);
             }
 
+            if (isset($postValues['smart_check']['value'])) {
+                $smartCheck = $postValues['smart_check']['value'];
+            } elseif (isset($postValues['smart_check']['inherit']) && $postValues['smart_check']['inherit'] == 1) {
+                $smartCheck = $this->dataHelper->getShopPw($storeId);
```

Comment:
Are you sure you want to store shop password in $smartCheck?

Reply:
fixed.